source/transfers/test/TransferHandlerRegistry.js

```diff
@@ -0,0 +1,1008 @@
+    it('Checks approvals (Alice 250 AST and 0 DAI, Bob 0 AST and 500 DAI)', async () => {
+      emitted(
+        await tokenAST.approve(swapAddress, 250, { from: aliceAddress }),
+        'Approval'
+      )
+      emitted(
+        await tokenOMG.approve(swapAddress, 200, { from: aliceAddress }),
+        'Approval'
+      )
+      emitted(
+        await tokenDAI.approve(swapAddress, 1000, { from: bobAddress }),
+        'Approval'
+      )
+      ok(
+        await allowances(aliceAddress, swapAddress, [
+          [tokenAST, 250],
+          [tokenDAI, 0],
+          [tokenOMG, 200],
+        ])
+      )
+      ok(
+        await allowances(bobAddress, swapAddress, [
+          [tokenAST, 0],
+          [tokenDAI, 1000],
+        ])
+      )
+    })
+  })
+
+  describe('Swaps (Fungible)', async () => {
+    let _order
+
+    before('Alice creates an order for Bob (200 AST for 50 DAI)', async () => {
+      _order = await orders.getOrder({
+        signer: {
+          wallet: aliceAddress,
+          token: tokenAST.address,
+          amount: 200,
+        },
+        sender: {
+          wallet: bobAddress,
+          token: tokenDAI.address,
+          amount: 50,
+        },
+      })
+
+      _order.signature = await signatures.getWeb3Signature(
+        _order,
+        aliceAddress,
+        swapAddress,
+        GANACHE_PROVIDER
+      )
+    })
+
+    it('Checks that Bob can swap with Alice (200 AST for 50 DAI)', async () => {
+      emitted(await swap(_order, { from: bobAddress }), 'Swap')
+    })
+
+    it('Checks balances...', async () => {
+      ok(
+        await balances(aliceAddress, [
+          [tokenAST, 800],
+          [tokenDAI, 50],
+        ]),
+        'Alice balances are incorrect'
+      )
+      ok(
+        await balances(bobAddress, [
+          [tokenAST, 200],
+          [tokenDAI, 950],
+        ]),
+        'Bob balances are incorrect'
+      )
+    })
+
+    it('Checks balances...', async () => {
+      ok(
+        await balances(aliceAddress, [
+          [tokenAST, 800],
+          [tokenDAI, 50],
+        ]),
+        'Alice balances are incorrect'
+      )
+      ok(
+        await balances(bobAddress, [
+          [tokenAST, 200],
+          [tokenDAI, 950],
+        ]),
+        'Bob balances are incorrect'
+      )
+    })
+
+    it('Checks that Alice cannot trade more than approved (200 AST)', async () => {
+      const order = await orders.getOrder({
+        signer: {
+          wallet: aliceAddress,
+          token: tokenAST.address,
+          amount: 200,
+        },
+      })
+
+      await reverted(swap(order, { from: bobAddress }))
+    })
+
+    it('Checks that Bob can not trade more than he holds', async () => {
+      const order = await orders.getOrder({
+        signer: {
+          wallet: bobAddress,
+          token: tokenDAI.address,
+          amount: 1000,
+        },
+        sender: {
+          wallet: aliceAddress,
```

Comment:
The default token for alice in this test is address(0) which would fail too. You should give a sender token and amount so that we know the failure is from bob's DAI not from alice's.